### PR TITLE
Adding request execution delay when placeholder mode in enabled in Kubernetes env

### PR DESCRIPTION
**Pull request by sushmithavangala**
Http triggers on lima ran in to a race condition where before the specialization of the app request was forwarded to the app. This PR fixes the same issue.

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Middleware/PlaceholderSpecializationMiddleware.cs [hunk 1]
```diff
@@ -28,6 +28,12 @@ public class PlaceholderSpecializationMiddleware
 
         public async Task Invoke(HttpContext httpContext)
         {
+            // delaying request execution
+            if (_webHostEnvironment.DelayRequestsEnabled)
+            {
```
Comment (pragnagopa) [inline src/WebJobs.Script.WebHost/Middleware/PlaceholderSpecializationMiddleware.cs]: change here will impact specialization code across other SKUs.  Please add conditional check EnvironmentCheckMiddleware  as discussed to ensure scope of the change is only for apps in K4Apps

Comment (fabiocav) [reply src/WebJobs.Script.WebHost/Middleware/PlaceholderSpecializationMiddleware.cs]: Indeed. This logic shouldn't be as the component we've discussed is responsible for this behavior. Can you please share more details about the motivation for this approach or problems you have found with the suggested change?

**Review by fabiocav — CHANGES_REQUESTED** [review]

File: src/WebJobs.Script.WebHost/Middleware/PlaceholderSpecializationMiddleware.cs (hunk 1 above)
Comment (sushmithavangala) [reply src/WebJobs.Script.WebHost/Middleware/PlaceholderSpecializationMiddleware.cs]: @pragnagopa  @fabiocav 

I was wondering if PlaceholderSpecializationMiddleware takes all the responsibility of handling specialization path. If EnvironmentCheckMiddleware is the middleware designed to handle this responsibility I’d change it to using Kubernetes env as an additional condition to use EnvironmentCheckMiddleware as we’ve discussed.

Comment (sushmithavangala) [reply src/WebJobs.Script.WebHost/Middleware/PlaceholderSpecializationMiddleware.cs]: @pragnagopa @fabiocav from https://github.com/Azure/azure-functions-host/blob/25c56c9f1d760a938857c553f9753ba15418d2c4/src/WebJobs.Script.WebHost/Middleware/EnvironmentReadyCheckMiddleware.cs#L11 understood that EnvironmentCheckMiddleware takes the role of delaying requests. Moved the logic to adding EnvironmentCheckMiddleware for kubernetes env as discussed before.

File: src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs [hunk 2]
```diff
@@ -36,7 +36,7 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
             builder.UseMiddleware<HttpRequestBodySizeMiddleware>();
             builder.UseMiddleware<SystemTraceMiddleware>();
             builder.UseMiddleware<HostnameFixupMiddleware>();
-            if (environment.IsLinuxConsumption())
+            if (environment.IsLinuxConsumption() || environment.IsKubernetesManagedHosting())
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs:39]: So Kubernetes environment is also using the Assign specialization code that activates the delay feature [here](https://github.com/Azure/azure-functions-host/blob/f1c6016b8e3a0f6ad8f02eb04297badf118e7acb/src/WebJobs.Script.WebHost/Management/InstanceManager.cs#L150)?

Comment (fabiocav) [reply src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs:39]: Correct.

**Review by fabiocav — APPROVED** [review]

**Review by mathewc — APPROVED** [review]

**Comment by liliankasem** [comment]
@SushmithaVReddy Looks like this has been approved, anything blocking this PR from being merged?

**Comment by sushmithavangala** [comment]
> @SushmithaVReddy Looks like this has been approved, anything blocking this PR from being merged?

@liliankasem I don't see a merge option for myself. Not sure If I have the access to do a merge.

**Comment by liliankasem** [comment]
> > @SushmithaVReddy Looks like this has been approved, anything blocking this PR from being merged?
> 
> @liliankasem I don't see a merge option for myself. Not sure If I have the access to do a merge.

Ah I see, just noticed this is a fork. We actually don't have a way to run our AzDO pipelines on forks right now; can you recreate this PR as a branch in the host repo? If not, I can find the time to do it later today

**Comment by sushmithavangala** [comment]
> > > @SushmithaVReddy Looks like this has been approved, anything blocking this PR from being merged?
> > 
> > 
> > @liliankasem I don't see a merge option for myself. Not sure If I have the access to do a merge.
> 
> Ah I see, just noticed this is a fork. We actually don't have a way to run our AzDO pipelines on forks right now; can you recreate this PR as a branch in the host repo? If not, I can find the time to do it later today

Thanks @liliankasem. I couldn't push my local branch to remote without forking.

**Comment by liliankasem** [comment]
Closing in favour of #8532

**Pull request closed without merging**